test/gen_casadi_test.py

```diff
@@ -1526,13 +1526,164 @@ def test_unspecified_dimensions(self):
                                      [3, 4],
                                      [5, 6]])
 
+    def test_wrong_unspecified_dimensions(self):
```

Comment:
This test did not catch the `dim_i` versus `val_dim_i` mismatch in the error message.